packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -760,3 +820,42 @@ export async function retryAsync<T>(fn: () => Promise<T>, retries: number, delay
   // Should never actually reach this, but for the sake of typescript
   throw new Error(`React a maximum of ${retries} retries.`);
 }
+
+/**
+ * @dev This function is a wrapper around the queryFilter function that splits the query into smaller chunks if the query is too large.
+ * @param contract - The contract to query.
+ * @param filter - The filter to apply to the query.
+ * @param fromBlock - The block number to start the query from.
+ * @param toBlock - The block number to end the query at.
+ * @returns The filtered events.
+ */
+export function queryFilterSafe(contract: Contract) {
```

Comment:
Very interesting strategy. This (along with the Across cache/retry provider) would be great for a universal provider library.